src/clr-angular/popover/tooltip/tooltip-trigger.ts

```diff
@@ -3,30 +3,28 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Directive, HostListener, Inject, Input } from '@angular/core';
+
+import { Directive, HostListener } from '@angular/core';
 import { IfOpenService } from '../../utils/conditional/if-open.service';
+import { TooltipIdService } from './providers/tooltip-id.service';
 import { Subscription } from 'rxjs';
-import { UNIQUE_ID } from '../../utils/id-generator/id-generator.service';
 
 @Directive({
   selector: '[clrTooltipTrigger]',
   host: {
     tabindex: '0',
     '[class.tooltip-trigger]': 'true',
-    '[attr.aria-describedby]': 'tooltipId',
-    '[attr.aria-label]': 'tooltipLabel',
-    role: 'button',
+    '[attr.aria-describedby]': 'ariaDescribedBy',
+    '[attr.role]': '"button"',
   },
 })
 export class ClrTooltipTrigger {
   public ariaDescribedBy;
   private subs: Subscription[] = [];
-  constructor(@Inject(UNIQUE_ID) public tooltipId: string, private ifOpenService: IfOpenService) {}
-
-  // This must be supplied by consumers
-  // This must be unique (among tooltips)
-  // This must be human readable e.g 'Tooltip 1' or 'Email input field tooltip`
-  @Input('clrTooltipLabel') tooltipLabel;
+  constructor(private ifOpenService: IfOpenService, private tooltipIdService: TooltipIdService) {
+    // The aria-described by comes from the id of content. It
+    this.subs.push(this.tooltipIdService.id.subscribe(idChange => (this.ariaDescribedBy = idChange)));
```

Comment:
Just for clarity could you change `idChange` to `tooltipId`?

Reply:
Yes. Will change.